android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -47,6 +47,13 @@ open class MainActivity : ReactActivity() {
         super.onNewIntent(intent)
     }
 
+    override fun onDestroy() {
```

Comment:
Before anything I have chosen to drop this commit and changes introduced by it. As to the reason for this - it is explained below

> What's the 'double initialization' bug?

From what I could infer double initialization bug is a bug where another instance of the app gets started while the previous one is running, this leads to sharing of the redux store that tends to introduce inconsistencies.
More discussion on this can be found at [CZO](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/double.20initialization.20on.20sharing)

**Possible steps of reproduction**
These are Listed [here](https://github.com/zulip/zulip-mobile/pull/4204#issuecomment-662295317)

I can't seem to run that branch locally (unrelated react native error) so I couldn't verify it there, but I did test it out on the rebased version of the app, at the point where I introduced initialParams [f0428a7](https://github.com/zulip/zulip-mobile/pull/4514/commits/f0428a7c36212fb665138cd18ca8289bfc1f3ebe)
I tried the steps of repro with the changes reverted and with the changes present. The symptoms reported were not noticed in the user interface.
However I did notice in debug console two symptoms that do correspond to this issue, occurrence of `action persist/REHYDRATE` as well as the log `Running "ZulipMobile" with {"rootTag":21}` two times.
These occurred when I reverted the changes introduced in 'fix re initialization issue` commit, and disappeared when the changes were present.
This was the reason I kept this commit before as it indicated double initialization.

**Why this specific change was supposed to fix this issue?**
It was discussed in [CZO](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/double.20initialization.20on.20sharing/near/950044)  that when the app were quit, it didn't fully quit, it was discovered that `react native host` was still present and on starting the app again in the way the repro steps demanded, the `react native host` would mount another react root view, which would trigger rehydrate and we would have a double initialization issue.

Except this may not be the complete story. When an app quits android doesn't kill it's process entirely, this is true for every app be it react app or not (possibly for optimization purposes for when the app is restarted). This is verifiable as the profiler still indicates the app process as alive even after the app's onDestroy method is triggered (Up until the point when the app is available in recent apps view.)

The Symptoms that we discover then are not because our app is reinitialized, while a previous instance is running; but instead when we do quit the app (and whenever it's onDestroy is called), it also destroys the `react root view` that houses our app without killing the `react native host` so then when we do bring the app to foreground once again a new `react root view` is mounted.
it is certain rootview is destroyed when onDestroy is called, I discovered this by following super.onDestroy by pressing `CTRL+B` in android studio, and it never kills react native host, but does unmount react root view. This was also discussed in [CZO](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/double.20initialization.20on.20sharing/near/950932)
This behavior is also in agreement with other react apps. `react native host` doesn't die but the root view is unmounted. So the double logs we were seeing is because root view was mounted before (it got unmounted without any log) and then mounts again. The dispatch to `rehydrate` happens after as well and we report it as having seen `rehydrate` twice, but at any one moment we only have 1 root view and 1 `rehydrate` dispatch.

**So what can be the cause of the user interface symptoms discovered in the reproduction step?**
I suspect it had something to do with implementation of `onNewIntent`. Checking in the dates of the repro comment and the commit editing `onNewIntent`. The comment comes before.
Why do I suspect it to be `onNewIntent`? because the implementation at the time of writing that commit was not the same as what it is today, and the repro steps trigger `onNewIntent` method. Although It seems a bit hard to verify, I will have to manually revert changes.
In any case the symptoms of double console log are also present in the current master as well - if one quits the app by pressing back button, and restarts the app from recent app, another rootView gets mounted while we can see a  `Running "ZulipMobile" with {"rootTag":21}` before.